File: Client/src/app/entity/distributionProduct.ts
import {Product} from "./Product";

export class DistributionProduct {
  id?: number;
  quantity?: number;
  product?: Product;
  lineTotal?: number;

  constructor(id?: number, product?: Product,  quantity?: number,lineTotal?: number) {
    this.id = id;
    this.quantity = quantity;
    this.product = product;
    this.lineTotal = lineTotal;
  }


}

File: Server/src/main/java/lk/wnl/wijeya/entity/DistributionProduct.java
package lk.wnl.wijeya.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "distribution_product")
public class DistributionProduct {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;
    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "distribution_id", nullable = false)
    private Distribution distribution;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(name = "quantity")
    private Integer quantity;

    @Column(name = "line_total", precision = 7, scale = 2)
    private BigDecimal lineTotal;

}